data-prepper-plugins/drop-events-processor/src/test/java/com/amazon/dataprepper/plugins/processor/drop/DropEventsProcessorTests.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.drop;
+
+import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.event.JacksonEvent;
+import com.amazon.dataprepper.model.record.Record;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class DropEventsProcessorTests {
+    private String messageInput;
+    private DropEventsProcessor dropProcessor;
+    private PluginSetting pluginSetting;
+    private final String PLUGIN_NAME = "drop_events";
+
+    @Test
+    void doExecute() {
+        pluginSetting = getDefaultPluginSetting();
+        pluginSetting.setPipelineName("dropProcessorPipeline");
+        dropProcessor = new DropEventsProcessor(pluginSetting);
+
+        final Map<String, Object> testData = new HashMap();
+        messageInput = UUID.randomUUID().toString();
+        testData.put("message", messageInput);
+        final Record<Event> record = buildRecordWithEvent(testData);
+
+        final List<Record<Event>> droppedRecords = (List<Record<Event>>) dropProcessor.doExecute(Collections.singletonList(record));
+
+        assertThat(droppedRecords.size(), equalTo(0));
+    }
+
+    @Test
+    void doExecuteMulti() {
+        pluginSetting = getDefaultPluginSetting();
+        pluginSetting.setPipelineName("dropProcessorPipeline");
+        dropProcessor = new DropEventsProcessor(pluginSetting);
+
+        final Map<String, Object> testData = new HashMap();
+        messageInput = UUID.randomUUID().toString();
+        testData.put("message", messageInput);
+        final Record<Event> record = buildRecordWithEvent(testData);
+
+        messageInput = UUID.randomUUID().toString();
+        testData.put("message", messageInput);
+        final Record<Event> record2 = buildRecordWithEvent(testData);
+
+        final List<Record<Event>> multiSet = new LinkedList<>();
```

Comment:
A couple convinience functions you could use to get a similar result:
`final List<Record<Event>> multiSet = List.of(record, record2);`
`final List<Record<Event>> multiSet = Collections.nCopies(3, record);`